r-build.md

```diff
@@ -0,0 +1,177 @@
+ 4. [Install additional R packages](#4-install-packages) 
+ 5. [Create a portable copy of your R installation that can be brought along with your jobs](#5-create-a-portable-copy-of-your-r-installation)
+ 6. [Use your custom R installation in your jobs](#6-use-your-custom-r-installation-in-your-jobs) 
+
+# Compile R
+
+## 1. Download R Source Code
+
+Before running any commands on CHTC, use a browser to get the source code 
+for your desired version of R from [CRAN](https://cran.r-project.org) 
+Under "Source Code for all Platforms", find the R-#.#.#.tar.gz file for your 
+desired version of R and download it to your computer before transferring a copy 
+to your `/home` directory on the submit server. 
+
+## 2. Submit An Interactive Job To A Build Server
+
+Creating an R installation can be computationally intensive and should not be 
+performed on the submit server. Instead, you will create your installation 
+on a CHTC build server by using an interactive job. 
+
+The interactive job is essentially a job without an executable; 
+you will be the one running the commands, instead (in this case, to install R).
+Like a regular HTCondor job, once you finish your R installation on the build server, 
+the output files (your completed portable R installation) will be transferred back to 
+the submit server (so that you can use the R installation for later jobs). 
+
+To submit an interactive job, create a submit file called `build.sub` as shown below:
+
+``` {.sub}
+# build.sub
+# Software build file
+
+universe = vanilla
+log = interactive.log
+
+# change the name of the file to be the name of your source code
+transfer_input_files = R-#.#.#.tar.gz
+
++IsBuildJob = true
+requirements = (OpSysMajorVer =?= 7)
+request_cpus = 1
+request_memory = 4GB
+request_disk = 4GB
+
+queue
+```
+
+Be sure to modify the above `transfer_input_files` statement with the appropriate 
+name of the R source code that you downloaded in the previous ste.
+
+Once this submit file is created, you will start the interactive job by
+running the following command:
+
+``` 
+[alice@submit]$ condor_submit -i build.sub
+```
+{:.term}
+
+The interactive build job should start in about a minute. Once it has
+started, the job has a time limit of four hours.
+
+## 3. Compile Base R From Source Code
+
+Once the interactive job starts, R can be compiled. The general workflow will be 
+to first run the R configuration script followed by `make` and `make install`. 
+
+To install R, first create a directory that will hold your R installation (e.g. `R/`) 
+, then extract all of the source code files from R-#.#.#.tar.gz and move into the source code directory:
+
+``` 
+[alice@build]$ mkdir R/
+[alice@build]$ tar xzf R-#.#.#.tar.gz
+[alice@build]$ cd R-#.#.#/
+```
+{:.term}
+
+From the source code directory, run the following commands. **The `make` step may take a while to 
+complete!**: 
+
+```
+[alice@build]$ ./configure --prefix=$_CONDOR_SCRATCH_DIR/R
```

Comment:
When I ran this on a build node (happened to be `build3000.chtc.wisc.edu`) I got this error:
```
checking whether PCRE support suffices... no
configure: error: PCRE2 library and headers are required, or use --with-pcre1 and PCRE >= 8.32 with UTF-8 support
```

Resolved by running with `--with-pcre1`, but have no idea what that actually means.

Reply:
UGH. which version of R?
We should just add those libraries/headers.

Reply:
Ack, should have mentioned: `4.0.1`, the latest release.

Reply:
Sorry for late reply - Eric just built some new copies of R for us and used the flag `--with-pcre1`

Reply:
what versions of R did @EUdds build? is this flag needed specifically for 4.0.0+?

Reply:
4.0.2. Some minimal googling says that we'll need some other flag to specify that pcre2 is on the system. So either way this part of the docs will need to change.

Reply:
added `--with-pcre1` to doc